cai_causal_graph/time_series_causal_graph.py

```diff
@@ -504,6 +505,7 @@ def extend_graph(
                             destination=lagged_destination_node,
                             edge_type=edge.get_edge_type(),
                             meta=edge.meta,
+                            validate=False,
```

Comment:
Are we safe doing this? I guess if there is a cycle that is OK as we want it in the extended graph. For example a valid minimal graph is X[t-1] -> X[t] -> Y[t] -> Z[t] -> X[t] if we allow cycles, which one can make by forcing that `validate=False` when adding Z[t] -> X[t] edge. 

We would at least need some unit tests for this now.  For CG and TSCG for what happens once someone has a graph in a state where they ignore validation. Not sure we are confident with how it will all work.

Reply:
Have added a warning in the docstrings that this shouldn't be done to introduce cycles, and we don't guarantee any behavior if someone does this.

Reply:
Would just add a comment here so we remember why we are setting it to `False`.